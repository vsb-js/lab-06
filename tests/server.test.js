import request from "supertest";

import { app } from "../src/server.js";

describe("Server", () => {
  test("GET /", async () => {
    await request(app)
      .get("/")
      .expect(200)
  });

  test(" GET /users", async () => {
    await request(app)
      .get('/users')
      .set('Accept', 'application/json')
      .expect('Content-Type', /json/)
      .expect(200)
      .then(response => {
        expect(response.text).toMatchSnapshot();
      })
  })
});