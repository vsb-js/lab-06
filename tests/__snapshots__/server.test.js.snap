// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server  GET /users 1`] = `"[{\\"id\\":1,\\"name\\":\\"Luke Skywalker\\",\\"height\\":172,\\"mass\\":77,\\"hair_color\\":\\"blond\\",\\"skin_color\\":\\"fair\\",\\"eye_color\\":\\"blue\\",\\"birth_year\\":\\"19BBY\\",\\"gender\\":\\"male\\"},{\\"id\\":2,\\"name\\":\\"C-3PO\\",\\"height\\":167,\\"mass\\":75,\\"hair_color\\":\\"n/a\\",\\"skin_color\\":\\"gold\\",\\"eye_color\\":\\"yellow\\",\\"birth_year\\":\\"112BBY\\",\\"gender\\":\\"n/a\\"},{\\"id\\":3,\\"name\\":\\"R2-D2\\",\\"height\\":96,\\"mass\\":32,\\"hair_color\\":\\"n/a\\",\\"skin_color\\":\\"white, blue\\",\\"eye_color\\":\\"red\\",\\"birth_year\\":\\"33BBY\\",\\"gender\\":\\"n/a\\"},{\\"id\\":4,\\"name\\":\\"Darth Vader\\",\\"height\\":202,\\"mass\\":136,\\"hair_color\\":\\"none\\",\\"skin_color\\":\\"white\\",\\"eye_color\\":\\"yellow\\",\\"birth_year\\":\\"41.9BBY\\",\\"gender\\":\\"male\\"},{\\"id\\":5,\\"name\\":\\"Leia Organa\\",\\"height\\":150,\\"mass\\":49,\\"hair_color\\":\\"brown\\",\\"skin_color\\":\\"light\\",\\"eye_color\\":\\"brown\\",\\"birth_year\\":\\"19BBY\\",\\"gender\\":\\"female\\"},{\\"id\\":6,\\"name\\":\\"Owen Lars\\",\\"height\\":178,\\"mass\\":120,\\"hair_color\\":\\"brown, grey\\",\\"skin_color\\":\\"light\\",\\"eye_color\\":\\"blue\\",\\"birth_year\\":\\"52BBY\\",\\"gender\\":\\"male\\"},{\\"id\\":7,\\"name\\":\\"Beru Whitesun lars\\",\\"height\\":165,\\"mass\\":75,\\"hair_color\\":\\"brown\\",\\"skin_color\\":\\"light\\",\\"eye_color\\":\\"blue\\",\\"birth_year\\":\\"47BBY\\",\\"gender\\":\\"female\\"},{\\"id\\":8,\\"name\\":\\"R5-D4\\",\\"height\\":97,\\"mass\\":32,\\"hair_color\\":\\"n/a\\",\\"skin_color\\":\\"white, red\\",\\"eye_color\\":\\"red\\",\\"birth_year\\":\\"unknown\\",\\"gender\\":\\"n/a\\"},{\\"id\\":9,\\"name\\":\\"Biggs Darklighter\\",\\"height\\":183,\\"mass\\":84,\\"hair_color\\":\\"black\\",\\"skin_color\\":\\"light\\",\\"eye_color\\":\\"brown\\",\\"birth_year\\":\\"24BBY\\",\\"gender\\":\\"male\\"},{\\"id\\":10,\\"name\\":\\"Obi-Wan Kenobi\\",\\"height\\":182,\\"mass\\":77,\\"hair_color\\":\\"auburn, white\\",\\"skin_color\\":\\"fair\\",\\"eye_color\\":\\"blue-gray\\",\\"birth_year\\":\\"57BBY\\",\\"gender\\":\\"male\\"}]"`;
